jib-plugins-common/src/test/java/com/google/cloud/tools/jib/plugins/common/JibBuildRunnerTest.java

```diff
@@ -215,4 +222,27 @@ public void testBuildImage_other()
       Assert.assertEquals(TEST_HELPFUL_SUGGESTIONS.none(), ex.getMessage());
     }
   }
+
+  @Test
+  public void testBuildImage_writesImageJson() throws Exception {
+    final ImageReference targetImageReference = ImageReference.parse("gcr.io/distroless/java:11");
+    final String imageId =
+        "sha256:61bb3ec31a47cb730eb58a38bbfa813761a51dca69d10e39c24c3d00a7b2c7a9";
+    final String digest = "sha256:3f1be7e19129edb202c071a659a4db35280ab2bb1a16f223bfd5d1948657b6fc";
+
+    final Path outputPath = temporaryFolder.newFile("jib-image.json").toPath();
+
+    Mockito.when(mockJibContainer.getTargetImage()).thenReturn(targetImageReference);
+    Mockito.when(mockJibContainer.getImageId()).thenReturn(DescriptorDigest.fromDigest(imageId));
+    Mockito.when(mockJibContainer.getDigest()).thenReturn(DescriptorDigest.fromDigest(digest));
+    Mockito.when(mockJibContainerBuilder.containerize(mockContainerizer))
+        .thenReturn(mockJibContainer);
+    testJibBuildRunner.writeImageJson(outputPath).runBuild();
+
+    final String outputJson = new String(Files.readAllBytes(outputPath), StandardCharsets.UTF_8);
```

Comment:
I think it would be better to compare this output explicitly as a string to ensure we're producing expected output.  (For example, to detect a situation where we change JSON libraries that produces slightly-invalid JSON.)

Reply:
Had a look at this. Beyond putting the full expected json into a string in the testcase (which seems highly specific) I didn't find a good way of doing that. 

As of now, the next line of `ImageMetadataOutput.fromJson(outputJson)` will at least make sure it properly parses into the expected type so it should catch any invalid JSON, serialization/deserialization differences and similar.